pkg/azclient/cloud.go

```diff
@@ -123,61 +136,35 @@ func AzureCloudConfigOverrideFromEnv(config *cloud.Configuration) (*cloud.Config
 			Audience: envConfig.TokenAudience,
 		}
 	}
-	return config, nil
+	if len(envConfig.StorageEndpointSuffix) > 0 {
+		armconfig.StorageSuffix = &envConfig.StorageEndpointSuffix
+	}
+	if len(envConfig.ContainerRegistryDNSSuffix) > 0 {
+		armconfig.ACRLoginServer = &envConfig.ContainerRegistryDNSSuffix
+	}
+	return nil
 }
 
 // GetAzureCloudConfig returns the cloud configuration for the given ARMClientConfig.
-func GetAzureCloudConfig(armConfig *ARMClientConfig) (*cloud.Configuration, error) {
+func BackfillAzureCloudConfig(armConfig *ARMClientConfig) (*cloud.Configuration, error) {
+	config := &cloud.AzurePublic
 	if armConfig == nil {
-		return &cloud.AzurePublic, nil
+		return config, nil
 	}
-	if armConfig.ResourceManagerEndpoint != "" {
-		return AzureCloudConfigFromURL(armConfig.ResourceManagerEndpoint)
+	config = AzureCloudConfigFromName(armConfig.Cloud)
+	if err := OverrideAzureCloudConfigFromMetadataService(armConfig, config); err != nil {
+		return nil, err
 	}
-
-	return AzureCloudConfigOverrideFromEnv(AzureCloudConfigFromName(armConfig.Cloud))
+	err := AzureCloudConfigOverrideFromEnv(armConfig, config)
```

Comment:
return config, AzureCloudConfigOverrideFromEnv(armConfig, config)

Reply:
The parameter will be modified here just like how json.unmarshal handles the parameters

Reply:
renamed the function